spinn_front_end_common/interface/abstract_spinnaker_base.py

```diff
@@ -1677,20 +1677,34 @@ def _do_load(self):
         inputs = dict(self._mapping_outputs)
         inputs["WriteMemoryMapReportFlag"] = (
             self._config.getboolean("Reports", "reports_enabled") and
-            self._config.getboolean("Reports", "write_memory_map_report")
+            self._config.getboolean("Reports", "write_memory_map_report") and
+            application_graph_changed
         )
 
+        if not application_graph_changed:
+            inputs["ExecutableTargets"] = self._last_run_outputs[
+                "ExecutableTargets"]
+            inputs["LoadedReverseIPTagsToken"] = self._last_run_outputs[
+                "LoadedReverseIPTagsToken"]
+            inputs["LoadedIPTagsToken"] = self._last_run_outputs[
+                "LoadedIPTagsToken"]
+            inputs["LoadedIPTagsToken"] = self._last_run_outputs[
+                "LoadedIPTagsToken"]
+            inputs["LoadedIPTagsToken"] = self._last_run_outputs[
+                "LoadedIPTagsToken"]
+
         algorithms = list()
 
         # add report for extracting routing table from machine report if needed
         # Add algorithm to clear routing tables and set up routing
-        if not self._use_virtual_board:
+        if not self._use_virtual_board and application_graph_changed:
             algorithms.append("RoutingSetup")
             # Get the executable targets
             algorithms.append("GraphBinaryGatherer")
 
-        if helpful_functions.read_config(
-                self._config, "Mapping", "loading_algorithms") is not None:
+        loading_algorithm = helpful_functions.read_config(
+            self._config, "Mapping", "loading_algorithms")
+        if loading_algorithm is not None and application_graph_changed:
             algorithms.extend(
                 self._config.get("Mapping", "loading_algorithms").split(","))
         algorithms.extend(self._extra_load_algorithms)
```

Comment:
Why are we not using the variable that we created a few lines above?

Reply:
`loading_algorithm` that is…

Reply:
We could just do `algorithms.extend(loading_algorithm.split(","))` and all would be good.

Reply:
good catch. i didnt notice when i did the refactor